src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonValueInfoOfT.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// todo
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class JsonValueInfo<T> : JsonTypeInfo<T>
+    {
+        /// <summary>
+        /// todo
+        /// </summary>
+        /// <param name="converter"></param>
+        /// <param name="options"></param>
+        public JsonValueInfo(
+            JsonConverter<T> converter,
+            JsonSerializerOptions? options) : base(typeof(T), options, ClassType.Value)
+        {
+            ConverterBase = converter;
+            PropertyInfoForClassInfo = CreatePropertyInfoForClassInfo(Type, Type, converter, Options);
+        }
+
+        /// <summary>
+        /// todo
+        /// </summary>
+        public void CompleteInitialization()
+        {
+            _isInitialized = true;
+
+            //todo: should we not add?
```

Comment:
What is the benefit of adding given all the generated TypeInfos have references to each other via their `JsonPropertyInfo`s?

Reply:
It allows direct usage of the type in the existing (De)Serialize methods without having to fall back to reflection\IL Emit, and also allows `options.GetConverter(type)` to return the type without reflection\IL Emit.